92-AppendixC.Rmd

```diff
@@ -0,0 +1,312 @@
+# Importing survey data into R
+
+The first step of any survey analysis is bringing the survey data into R. This process is often referred to as importing, loading, or reading in data. Survey files come in different formats depending on the software used to create them. One of the many advantages of R is the flexibility in handling various data formats, regardless of their file extensions. Here are examples of common public-use survey file formats we may encounter:
+
+* Delimiter-separated text files
+* Excel spreadsheets in `.xls` or `.xlsx` format
+* R native `.rda` files
+* Stata datasets in `.dta` format
+* SAS datasets in `.sas` format
+* SPSS datasets in `.sav` format
+* Application Programming Interfaces (APIs), often in JSON format
+
+This appendix guides analysts through the process of importing various types of survey data into R. R also offers dedicated packages such as {googlesheets4} for Google Sheets or {qualtRics} for Qualtrics. With less common or proprietary file formats, the broader data science community can often provide guidance. Online resources like Stack Overflow and dedicated forums are valuable sources of information for importing data into R.
+
+## Importing delimiter-separated files into R
+
+Delimiter-separated files use specific characters, known as delimiters, to separate values within the file. For example, CSV (Comma-Separated Values) files use commas as delimiters, while TSV (Tab-Separated Values) files use tabs. These file formats are widely used because of their simplicity and compatibility with various software applications.
+
+The {readr} package, part of the tidyverse ecosystem, offers efficient ways to import delimiter-separated files into R. It provides several advantages, including automatic data type detection and flexible handling of missing values, depending on one's survey research needs. The {readr} package includes functions for:
+
+* `read_csv()`: This function is specifically designed to read CSV files.
+* `read_tsv()`: Use this function for Tab-Separated Values (TSV) files.
+* `read_delim()`: This function can handle a broader range of delimiter-separated files, including CSV and TSV. Specify the delimiter using the `delim` argument.
+* `read_fwf()`: This function is useful for reading Fixed-Width Files, where columns have predetermined widths, and values are aligned in specific positions.
+* `read_table()`: Use this function when dealing with whitespace-separated files, such as those with spaces or multiple spaces as delimiters.
+* `read_log()`: This function can read and parse web log files.
+
+In the example below, we use {readr} to load a CSV file named 'anes_timeseries_2020_csv_20220210.csv' into an R object called `anes_csv`. The `read_csv()` imports the file and stores the data in the `anes_csv` object. We can then use this object for further analysis.
+
+```r
+library(readr)
+
+anes_csv <-
```

Comment:
In the excel section you have the arguments for the function and explain it a bit more. Should we do that here too?  Most of these have similar arguments, so you could focus on maybe just csv and say that others are similar.

Reply:
Added!